### PR TITLE
deps: support OTel tracing v1.25.0

### PR DESCRIPTION
**Pull request by inge4pres**
### Reason for this PR

Addresses https://github.com/mailgun/holster/issues/189.

### Details

Minor version upgrades to the OTel suites and implement the interface in `DummySpan`.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,13 +1,15 @@
 module github.com/mailgun/holster/v4
 
-go 1.20
+go 1.21
```
Comment (vtopc) [inline go.mod:3]: https://github.com/mailgun/holster/pull/186#discussion_r1502649820

Comment (Baliedge) [reply go.mod:3]: Are we ready to drop 1.20 now?  Legacy codebases would have to stay at legacy `holster` releases.  Otherwise, we won't be able to update OTel, which is now set at 1.21.

e.g.:
```sh
$ go get go.opentelemetry.io/otel
go: upgraded go 1.20 => 1.21
go: added toolchain go1.21.5
go: upgraded github.com/go-logr/logr v1.3.0 => v1.4.1
go: upgraded github.com/stretchr/objx v0.5.0 => v0.5.2
go: upgraded github.com/stretchr/testify v1.8.4 => v1.9.0
go: upgraded go.opentelemetry.io/otel v1.21.0 => v1.25.0
go: upgraded go.opentelemetry.io/otel/metric v1.21.0 => v1.25.0
go: upgraded go.opentelemetry.io/otel/trace v1.21.0 => v1.25.0
```

And since 1.20 is now EOL, we should be justified.  https://endoflife.date/go

Comment (MatthewEdge) [reply go.mod:3]: If existing services can update without issue then dropping 1.20 is fine. GH Code Search to find the offenders and notify as necessary 👍

File: go.mod [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+
+toolchain go1.22.2
```
[the remark this answers is not in the document]
Comment (Baliedge) [reply go.mod:5]: Yes, let's use latest toolchain 1.21.x minor version.

[the remark this answers is not in the document]
Comment (Baliedge) [reply go.mod:5]: Actually, it seems `go mod tidy` is setting this and bumps `toolchain` to latest installed Golang.  We may not have a choice here.

**Comment by Baliedge** [comment]
@inge4pres I've recently merged https://github.com/mailgun/holster/pull/194, which addresses updating Golang version to 1.21.  Please update from `master` branch.  Then we can merge this.

Thanks for your contribution.

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (inge4pres) [reply go.mod:5]: yeah unfortunately the toolchain is needed to compile the updated dependency...

[the remark this answers is not in the document]
Comment (inge4pres) [reply go.mod]: note that the go requirement still is for 1.21, but then the toolchain used for compiling will be 1.22.
does this require to update the Github action workflows?
what is the reason to not require the latest Go version?

File: tracing/tracing.go [hunk 3]
```diff
@@ -18,9 +18,10 @@ import (
 	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
 	"go.opentelemetry.io/otel/trace"
 
-	"github.com/mailgun/holster/v4/errors"
 	"github.com/sirupsen/logrus"
 	"github.com/uptrace/opentelemetry-go-extra/otellogrus"
+
```
Comment (Baliedge) [inline tracing/tracing.go:23]: nit: Should revert this unnecessary change.

**Review by Baliedge — CHANGES_REQUESTED** [review]

File: tracing/tracing.go (hunk 3 above)
Comment (inge4pres) [reply tracing/tracing.go:23]: ah thanks, that's my IDE formatting imports, will do

Comment (Baliedge) [reply tracing/tracing.go:23]: Oh, I see.  It placed organizational imports in a 3rd section.  That's a new thing with Golang.  So it's not a mistake.

**Review by Baliedge — APPROVED** [review]

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (vtopc) [reply go.mod]: IDK, I can't reproduce this:
```
go version && go mod edit -go=1.21 && go mod tidy
go version go1.22.2 darwin/amd64
```

**Review by Baliedge — CHANGES_REQUESTED** [review]
The test is failing due to a required change that goes with the OTel update.  Apply this patch:

```
diff --git a/tracing/tracing.go b/tracing/tracing.go
index 7306a70..9831fb4 100644
--- a/tracing/tracing.go
+++ b/tracing/tracing.go
@@ -15,7 +15,7 @@ import (
        "go.opentelemetry.io/otel/propagation"
        "go.opentelemetry.io/otel/sdk/resource"
        sdktrace "go.opentelemetry.io/otel/sdk/trace"
-       semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+       semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
        "go.opentelemetry.io/otel/trace"

        "github.com/mailgun/holster/v4/errors"
```

**Comment by inge4pres** [comment]
> Apply this patch:

Thanks! 🙏🏼 just pushed

**Review by Baliedge — APPROVED** [review]
👍🏻

**Comment by inge4pres** [comment]
Thanks for the guidance @Baliedge 🙏🏼

File: tracing/tracing.go (hunk 3 above)
Comment (vtopc) [reply tracing/tracing.go:23]: Yeah, but there are 4 sections now.
